### PR TITLE
introduce stage:add command

### LINKED ISSUE #5253: dvc: introduce dvc stage add(create)
As the title says, we have to implement `dvc stage {add, create}` that will allow us to create a stage.

### PR DESCRIPTION
**Pull request by skshetry**
The changes include:
* Introduce new `stage:add` command
* Move common args to `stage:add` from `run`.
* Move most of the `repo.run` to `stage.utils.create_stage_from_cli`.
* Use `-C` option for checkpoints instead of `-c`. `-c` is now used in command for `stage:add`.
* Parse `--live-no-summary` and `--live-no-report` later in the `utils` than upfront.

Closes #5253 
* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: dvc/command/stage.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+    stage_add_parser.add_argument(
+        "--outs-persist-no-cache",
+        action="append",
+        default=[],
+        help="Declare output file or directory that will not be "
+        "removed upon repro (do not put into DVC cache).",
+        metavar="<filename>",
+    )
+    stage_add_parser.add_argument(
+        "--checkpoints",
```
Comment (skshetry) [inline dvc/command/stage.py:163]: I removed the short flag for `checkpoints` and used it for the command. I'll update `run` after this PR gets merged.

File: dvc/command/stage.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+class CmdStageAdd(CmdBase):
+    def run(self):
+        from dvc.stage.utils import create_stage_from_cli
+
+        kwargs = vars(self.args)
+
+        cmd = kwargs.pop("command")
+        assert len(cmd) >= 1
+        kwargs["cmd"] = cmd[0] if len(cmd) == 1 else cmd
```
Comment (skshetry) [inline dvc/command/stage.py:18]: Supporting multiple commands now. But, we should have an opinion here: either a multiline string or a list of cmds.

Comment (efiop) [reply dvc/command/stage.py:18]: list sounds better to me :)

File: dvc/command/run.py [hunk 3]
```diff
@@ -43,8 +42,8 @@ def run(self):
                 plots=self.args.plots,
                 plots_no_cache=self.args.plots_no_cache,
                 live=self.args.live,
-                live_summary=not self.args.live_no_summary,
-                live_report=not self.args.live_no_report,
+                live_no_summary=self.args.live_no_summary,
+                live_no_report=self.args.live_no_report,
```
Comment (skshetry) [inline dvc/command/run.py:46]: Moved this negative logic to `utils.create_stage_from_cli` helper.

File: dvc/command/run.py [hunk 4]
```diff
@@ -91,6 +90,8 @@ def _quote_argument(self, argument):
 
 
 def add_parser(subparsers, parent_parser):
+    from dvc.command.stage import _add_common_args
```
Comment (skshetry) [inline dvc/command/run.py:93]: Sharing common args between `stage:add` and `run`.

File: dvc/repo/run.py [hunk 5]
```diff
@@ -20,27 +18,10 @@ def run(
 ):
     from dvc.stage.utils import check_graphs, create_stage_from_cli
 
-    if not kwargs.get("cmd"):
```
Comment (skshetry) [inline dvc/repo/run.py:23]: Moved all of this to utilities.

**Review by pared — APPROVED** [review]

File: dvc/repo/run.py [hunk 6]
```diff
@@ -20,27 +18,10 @@ def run(
...
-
-    if not stage_name and not single_stage:
-        raise InvalidArgumentError("`-n|--name` is required")
-
     stage = create_stage_from_cli(
         self, single_stage=single_stage, fname=fname, **kwargs
     )
+
     if kwargs.get("run_cache", True) and stage.can_be_skipped:
         return None
```
Comment (skshetry) [inline dvc/repo/run.py:26]: @efiop, let me know if we want to remove this behaviour of `Stage is cached, skipping` in 2.0. If so, I'll get rid of it in the next PR.

File: dvc/command/stage.py (hunk 1 above)
Comment (pmrowla) [reply dvc/command/stage.py:136]: I think we might still want to keep a short option for `--checkpoints`, maybe use capital `-C` now?

@dmpetrov

Comment (skshetry) [reply dvc/command/stage.py:137]: @pmrowla, introduce `-C`. Thanks.

**Review by pmrowla — APPROVED** [review]

File: dvc/repo/run.py (hunk 6 above)
Comment (efiop) [reply dvc/repo/run.py:26]: @skshetry Yeah, could totally drop it or refactor it. The functionality is partially covered by run-cache, but not fully. But this `can_be_skipped` is flawed by itself :(

File: dvc/command/stage.py [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+        "removed upon repro (do not put into DVC cache).",
+        metavar="<filename>",
+    )
+    parser.add_argument(
+        "-C",
+        "--checkpoints",
+        action="append",
+        default=[],
+        help=argparse.SUPPRESS,
+        metavar="<filename>",
```
Comment (jorgeorpinel) [inline dvc/command/stage.py:141]: Hi! This seems to be hidden. Just wondering whether there are plans to enable it, per iterative/dvc.org/issues/2253

Comment (pmrowla) [reply dvc/command/stage.py:141]: Likely a leftover from when experiments were opt-in pre-2.0 release, it should no longer be suppressed. Will submit a fix for this later today

**Pull request merged**